### PR TITLE
feat(connector): [Adyen] Implement Alma BNPL and DANA Wallet

### PR DESCRIPTION
**Pull request by Sakilmostak**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

### Additional Changes

- [x] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

Alma and DANA was not available as a payment method so I add it.

## How did you test it?

Alma Testing:
Screen Shot 2023-06-29 at 12 10 04 PM
Screen Shot 2023-06-29 at 12 12 02 PM
Screen Shot 2023-06-29 at 12 12 07 PM
Screen Shot 2023-06-29 at 12 12 12 PM
Note: Alma requires OTP from a valid mobile number in redirection to accept payment so automation test is not written

Dana Testing: 
9CF20419-883D-4CE0-837E-818EE4E40837_1_201_a
Screen Shot 2023-06-29 at 11 46 59 AM
Screen Shot 2023-06-29 at 11 48 28 AM
Screen Shot 2023-06-29 at 11 48 35 AM
Screen Shot 2023-06-29 at 11 48 41 AM

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [x] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/tests/connectors/adyen_uk_ui.rs [hunk 1]
```diff
@@ -118,6 +118,35 @@ async fn should_make_adyen_twint_payment(driver: WebDriver) -> Result<(), WebDri
     Ok(())
 }
 
+async fn should_make_adyen_dana_payment(driver: WebDriver) -> Result<(), WebDriverError> {
```
Comment (ArjunKarthik) [inline crates/router/tests/connectors/adyen_uk_ui.rs:121]: Can’t we have ui test for Alma?

**Review by ArjunKarthik — CHANGES_REQUESTED** [review]

**Review by ArjunKarthik — APPROVED** [review]

File: crates/router/tests/connectors/adyen_uk_ui.rs (hunk 1 above)
Comment (Sakilmostak) [reply crates/router/tests/connectors/adyen_uk_ui.rs:172]: Alma requires OTP from mobile number for validation so can't be automated

File: crates/router/tests/connectors/adyen_uk_ui.rs [hunk 2]
```diff
@@ -192,6 +192,35 @@ async fn should_make_adyen_twint_payment(driver: WebDriver) -> Result<(), WebDri
...
+    let conn = AdyenSeleniumTest {};
+    conn.make_redirection_payment(
+        driver,
+        vec![
+            Event::Trigger(Trigger::Goto(&format!("{CHEKOUT_BASE_URL}/saved/175"))),
+            Event::Trigger(Trigger::Click(By::Id("card-submit-btn"))),
+            Event::Trigger(Trigger::SendKeys(
+                By::XPath("/html/body/div[1]/div[2]/div[1]/div/input"),
```
Comment (jagan-jaya) [inline crates/router/tests/connectors/adyen_uk_ui.rs:203]: Prefer using a unique ID or class or minimum xpath , if the website updates with new structure, this test will fail. Only go with full Xpath when u can't find any of the above.

**Review by jagan-jaya — APPROVED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Review by ArjunKarthik — APPROVED** [review]

File: crates/router/tests/connectors/adyen_uk_ui.rs (hunk 2 above)
Comment (Sakilmostak) [reply crates/router/tests/connectors/adyen_uk_ui.rs]: Done👍

**Review by ArjunKarthik — APPROVED** [review]

**Review by srujanchikke — DISMISSED** [review]

**Review by ashokkjag — APPROVED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Review by srujanchikke — APPROVED** [review]

**Review by ashokkjag — APPROVED** [review]

**Pull request merged**